content/v2.0/write-data/best-practices/optimize-writes.md

```diff
@@ -50,6 +55,53 @@ For better performance, use the coarsest precision possible for timestamps.
 
 _Specify timestamp precision when [writing to InfluxDB](/v2.0/write-data/#timestamp-precision)._
 
+## Use gzip compression
+
+Enable gzip compression when writing to InfluxDB to speed up data ingestion.
+Benchmarks have shown up to a 5x speed improvement when data is compressed.
+
+{{< tabs-wrapper >}}
+{{% tabs %}}
+[Telegraf](#)
+[Client libraries](#)
+[InfluxDB API](#)
+{{% /tabs %}}
+{{% tab-content %}}
+### Enable gzip compression in Telegraf
+
+In the `influxdb_v2` output plugin configuration in your `telegraf.conf`, set the
+`content_encoding` option to `gzip`:
+
+```toml
+[[outputs.influxdb_v2]]
+  urls = ["http://localhost:9999"]
+  # ...
+  content_encoding = "gzip"
+```
+{{% /tab-content %}}
+{{% tab-content %}}
+### Enable gzip compression in InfluxDB client libraries
```

Comment:
Curious, do we know which libraries enforce compression by default? I looked at a couple of our client libraries and didn't see mention of compression or gzip (but may be looking in wrong place).

Reply:
Each is different, but you can find compression implementations in each library's source code. We should probably have bonitoo take another pass at the client library readmes and document all of the available options.